tools/setup_helpers/generate_code.py

```diff
@@ -79,6 +77,85 @@ def generate_code(ninja_global=None,
             python_install_dir,
             autograd_dir)
 
+def get_selector_from_legacy_operator_selection_list(
+        selected_op_list_path: str,
+):
+    from tools.autograd.utils import load_op_list_and_strip_overload
+
+    selected_op_list = load_op_list_and_strip_overload(
+        None,
+        selected_op_list_path,
+    )
+
+    is_root_operator = False
+    is_used_for_training = False
+
+    if selected_op_list_path is not None:
+        # This is a temporary hack (which will be cleaned up very soon)
+        # to be able to determine which invocations are for what type
+        # of operation. We force the use of the file name to determine
+        # which use-case is being catered to for this specific
+        # invocation of generate_code.py
+        #
+        # For the internal build, we have fixed things up to only use
+        # the file name based operator selection strategy, and the open
+        # source version doesn't seem to use --selected-op-list at all
+        # and uses --selected-op-list path instead.
+        #
+        # ./caffe2/CMakeLists.txt:
+        # $<$<BOOL:${SELECTED_OP_LIST}>:--selected-op-list-path="${SELECTED_OP_LIST}">
+        #
+        # Please help me verify the following claims:
+        # -------------------------------------------
+        # 1. Currently, OSS doesn't support on-device learning, so there is
+        # no support for VariableType_N.cpp code-gen in OSS
+        #
+        # 2. OSS uses the full closure for unboxing wrappers code-gen, so
+        # it generates more unboxing wrappers than needed compared to
+        # the internal build.
+        #
+        # If we are in internal FB build, then we will expect one of the 2
+        # file paths below.
+        if "root_op_list.yaml" in selected_op_list_path or "combined_op_list.yaml" in selected_op_list_path:
+            is_root_operator = "root_op_list.yaml" in selected_op_list_path
+            is_used_for_training = "combined_op_list.yaml" in selected_op_list_path
```

Comment:
For internal BUCK build, do we call this codegen multiple times to generate unboxing_wrapper, variable_kernels, respectively?
If for unboxing_wrapper pass, we only use the unboxing_wrapper output and ignore variable_kernels output, and for variable_kernels pass, we only use the variable_kernels output and ignore unboxing_wrapper output, then does it matter whether is_root_operator / is_used_for_training is set precisely?

Can we simply set both to true as the OSS setup?

Reply:
+1 @ljk53  -  I think that this makes total sense - can merge the 2 - thanks for the simplification!

Reply:
Merged.